solver/src/solver/http_solver/settlement.rs

```diff
@@ -54,18 +53,26 @@ impl ExecutedLimitOrder {
     }
 }
 
-struct ExecutedConstantProductAmms {
+#[derive(Clone, Debug)]
+struct ExecutedConstantProductAmm {
     order: ConstantProductOrder,
     input: (H160, U256),
     output: (H160, U256),
 }
 
-struct ExecutedWeightedProductAmms {
+#[derive(Clone, Debug)]
+struct ExecutedWeightedProductAmm {
     order: WeightedProductOrder,
     input: (H160, U256),
     output: (H160, U256),
 }
 
+#[derive(Clone, Debug)]
+enum ExecutedAmm {
+    CP(ExecutedConstantProductAmm),
+    WP(ExecutedWeightedProductAmm),
```

Comment:
We could move the whole struct declaration here instead of keeping it separate:

```rust
enum ExecutedAmm {
    ExecutedConstantProductAmm {
        ...
    },
    ExecutedWeightedProductAmm { ... }
}
```

Not necessary just something I noticed because we never use the structs standalone.

Reply:
Cool, I wasn't aware of this! Will give it a try.

Reply:
Alternatively you could do this:

```rust
struct ExecutedAmm {
    input: (H160, U256),
    output: (H160, U256),
    order: ExecutedOrder
}

enum ExecutedOrder {
    ConstantProduct(ConstantProductOrder),
    WeightedProduct(WeightedProductOrder),
}
```

Which would help with your comment below that you need identical code in two match branches. With this struct input and output are always available.

Reply:
I had also tried this in the past, but there was something about it that wasn't working well. However I will try your second suggestion now. Since I am having trouble accessing the fields of the structs with your first suggestion.